aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -754,29 +759,58 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         TODO("Not yet implemented")
     }
 
-    override fun getCurrentUser(): AuthUser? {
-        var authUser: AuthUser? = null
-        val semaphore = Semaphore(0)
+    override fun getCurrentUser(
+        onSuccess: Consumer<AuthUser>,
+        onError: Consumer<AuthException>
+    ) {
         authStateMachine.getCurrentState { authState ->
-            when (val authorizationState = authState.authNState) {
+            when (authState.authNState) {
                 is AuthenticationState.SignedIn -> {
-                    authUser = AuthUser(
-                        authorizationState.signedInData.userId,
-                        authorizationState.signedInData.username
+                    var token: StateChangeListenerToken? = null
+                    token = credentialStoreStateMachine.listen(
+                        {
+                            when (it) {
+                                is CredentialStoreState.Success -> {
+                                    val accessToken = it.storedCredentials?.cognitoUserPoolTokens?.accessToken ?: ""
```

Comment:
is it necessary to use `""` instead of just `null`? wouldn't checking `isNullorEmpty` give you same result?